pkg/registry/apps/daemonset/strategy.go

```diff
@@ -68,6 +69,25 @@ func (daemonSetStrategy) NamespaceScoped() bool {
 	return true
 }
 
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+func (daemonSetStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
```

Comment:
How many times are these functions called? It would be a shame to construct this map for every request.

Reply:
They're called at initialization time, once.

Reply:
For builtin types it gets called by the endpoints installers so it shouldn't be constructed for every request, right?

https://github.com/kubernetes/kubernetes/pull/99661/files#diff-a8da7b9bda33b6055a1f5deb7d0c33058c5d706ec5b44c665191a359a43160c6R265 

And for CRDs it's built by the scope which is called in `getOrCreateServingInfoFor`, so again I don't think it's on every request.

https://github.com/kubernetes/kubernetes/pull/99661/files#diff-01e7a2059fe62e95ec32e2589c01b7218100de327647c641c334d38f2c5aa06eR967

Reply:
Actually, good point Kevin: "getOrCreateServingInfoFor" is (surprisingly) called for every request!